react-validated-input-component/src/validated-input.jsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+
+class ValidatedInput extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      val: '',
+      isPassed: false
+    };
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(e) {
+    const currentVal = e.target.value;
+    this.setState({
+      val: currentVal
+    });
+    if (
+      currentVal.length >= 8 &&
+      currentVal.match(/\d+/g) &&
+      currentVal.match(/.*[A-Z]/g) &&
+      currentVal.match(/[!@#$%^&*()]/g)
+    ) {
+      this.setState({ isPassed: true });
+    } else {
+      this.setState({ isPassed: false });
```

Comment:
Ok, for this challenge I recommend storing the error message in state so that you don't need to put the Regular Expression checks in two places.